libs/server/Resp/AdminCommands.cs

```diff
@@ -711,35 +711,62 @@ void ProcessHelloCommand(byte? respProtocolVersion, ReadOnlySpan<byte> username,
         }
 
         /// <summary>
-        /// Performs @admin command group permission checks for the current user and the given command.
-        /// (NOTE: This function is temporary until per-command permissions are implemented)
+        /// Performs permission checks for the current user and the given command.
+        /// (NOTE: This function does not check keyspaces)
         /// </summary>
-        /// <param name="bufSpan">Buffer containing the current command in RESP3 style.</param>
+        /// <param name="cmd">Command be processed</param>
+        /// <param name="ptr">Pointer to start of arguments in command buffer</param>
         /// <param name="count">Number of parameters left in the command specification.</param>
         /// <param name="processingCompleted">Indicates whether the command was completely processed, regardless of whether execution was successful or not.</param>
         /// <returns>True if the command execution is allowed to continue, otherwise false.</returns>
-        bool CheckACLAdminPermissions(ReadOnlySpan<byte> bufSpan, int count, out bool processingCompleted)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        bool CheckACLPermissions(RespCommand cmd, byte* ptr, int count, out bool processingCompleted)
         {
-            Debug.Assert(!_authenticator.IsAuthenticated || (_user != null));
+            Debug.Assert(_user != null);
 
-            if (!_authenticator.IsAuthenticated || (!_user.CanAccessCategory(CommandCategory.Flag.Admin)))
+            if (!_user.CanAccessCommand(cmd))
```

Comment:
I think this needs to be:
```
if ((!_authenticator.IsAuthenticated || !_user.CanAccessCommand(cmd)) && !cmd.IsNoAuth())
```

Assuming:
* if Auth fails, we want to proceed ONLY IF the command is a NoAuth command
* if ACL fails, we want to proceed ONLY IF the command is a NoAuth command

This means the NoAuth check has to be in the `&&` clause.

Reply:
Also note that `_authenticator.IsAuthenticated` needs to run no matter what code path, which is why it is the first check in the `if` condition. That way we are protected by both Auth and ACL in one place.